src/main/java/org/opensearch/flowframework/transport/ProvisionWorkflowTransportAction.java

```diff
@@ -123,6 +123,7 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
                 Workflow provisionWorkflow = template.workflows().get(PROVISION_WORKFLOW);
                 List<ProcessNode> provisionProcessSequence = workflowProcessSorter.sortProcessNodes(provisionWorkflow, workflowId);
                 workflowProcessSorter.validateGraph(provisionProcessSequence);
+                workflowProcessSorter.validatePluginsInstalled(provisionProcessSequence);
```

Comment:
There's some duplication in these two methods.  Suggest combining these into a single method call, then you only have to call `WorkflowValidator validator = WorkflowValidator.parse("mappings/workflow-steps.json")` once and can pass that object to the two individual methods.

Reply:
I intentionally left this separate from graph validation since we use graph validation when creating a workflow with the dry-run flag set to true. I can combine these, but then we would have installed plugin validation during create instead of only provision, which goes against the requirement in the issue. 

I can combine both if we're ok with having installed plugin validation for both APIs

Reply:
I think it's reasonable to validate both.

Reply:
sure ill make this change

Reply:
I have confirmed that installed plugin validation works for creating a workflow with validation (using an incorrect required plugin name for the create connector step) : 
```
curl -i -XPOST "localhost:9200/_plugins/_flow_framework/workflow?dryrun=true" -H "Content-Type:application/json" --data '{"name":"create-connector-register-deploy-model","description":"test case","use_case":"TEST_CASE","version":{"template":"1.0.0","compatibility":["2.12.0","3.0.0"]},"workflows":{"provision":{"nodes":[{"id":"workflow_step_1","type":"create_connector","user_inputs":{"name":"OpenAI Chat Connector","description":"The connector to public OpenAI model service for GPT 3.5","version":"1","protocol":"http","parameters":{"endpoint":"api.openai.com","model":"gpt-3.5-turbo"},"credential":{"openAI_key":"12345"},"actions":[{"action_type":"predict","method":"POST","url":"https://${parameters.endpoint}/v1/chat/completions"}]}},{"id":"workflow_step_2","type":"register_remote_model","previous_node_inputs":{"workflow_step_1":"connector_id"},"user_inputs":{"name":"openAI-gpt-3.5-turbo","function_name":"remote","description":"test model"}},{"id":"workflow_step_3","type":"deploy_model","previous_node_inputs":{"workflow_step_2":"model_id"}}],"edges":[{"source":"workflow_step_1","dest":"workflow_step_2"},{"source":"workflow_step_2","dest":"workflow_step_3"}]}}}'

HTTP/1.1 400 Bad Request
content-type: application/json; charset=UTF-8
content-length: 107

{"error":"The workflowStep create_connector requires the following plugins to be installed : [testsesete]"}
```